### PR TITLE
[Framework] Replace ZF1 email validator with Laminas

### PR DESCRIPTION
**Pull request by Den4ik**
### Description (*)

This PR replace usage of ZF1 email validator with Laminas library

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@Den4ik it is a backward incompatible change. I think we should create a separate interface and implementation for that. What do you think about it?

**Comment by Den4ik** [comment]
> @Den4ik it is a backward incompatible change. I think we should create a separate interface and implementation for that. What do you think about it?

@ihor-sviziev Totally agree with you and already working on it

File: lib/internal/Magento/Framework/Validator/EmailAddress.php [hunk 1]
```diff
@@ -5,9 +5,13 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
+
 namespace Magento\Framework\Validator;
 
-class EmailAddress extends \Zend_Validate_EmailAddress implements \Magento\Framework\Validator\ValidatorInterface
+use Laminas\Validator\EmailAddress as LaminasEmailAddress;
```
Comment (andrewbess) [inline lib/internal/Magento/Framework/Validator/EmailAddress.php]: Hello @Den4ik 
I think it should be EmailAddressValidator

```suggestion
use Laminas\Validator\EmailAddress as EmailAddressValidator;
```

What do you think about this?

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @Den4ik 
Thank you for your contribution
Please see my comments
Also, please remove redundant strings from copyrights
Thank you in advance.

**Comment by ihor-sviziev** [comment]
I closed this PR due to inactivity. Feel free to reopen if you wish to continue

**Comment by Den4ik** [comment]
> @Den4ik it is a backward incompatible change. I think we should create a separate interface and implementation for that. What do you think about it?

@ihor-sviziev I have checked dependencies and don't see possible fast solutions of using new interface due to usage at constructors of classes marked as `@api`. For example, https://github.com/magento/magento2/blob/2.4-develop/lib/internal/Magento/Framework/Validator.php

I believe that it could be skipped in this case by architect approval.

cc @sidolov

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
Close this PR due to multiple requirement of multiple changes for `@api` marked classes.
This thing will could be implemented when Magento will plan 2.5 release, but unfortunately it's not make sense for now

**Pull request closed without merging**